ablySpec/RealtimeClientChannel.swift

```diff
@@ -550,6 +550,40 @@ class RealtimeClientChannel: QuickSpec {
 
             }
 
+            // RTL7
+            context("subscribe") {
+
+                // RTL7c
+                it("should implicitly attach the channel") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    let channel = client.channels.get("test")
+
+                    channel.subscribe { _, _ in }
+
+                    expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+                }
+
+                // RTL7c
+                pending("should result in an error if channel is in the FAILED state") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    let channel = client.channels.get("test")
+                    channel.onError(AblyTests.newErrorProtocolMessage())
+                    expect(channel.state).to(equal(ARTRealtimeChannelState.Failed))
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.subscribe { message, errorInfo in
```

Comment:
Your interpretation of whether to raise an exception or pass an error in a listener is 100% correct.  Paddy, Simon and I [debated this for some time](https://github.com/ably/ably-js/commit/1b151df5558339d8b2fd80bbadd2375d9a138b4a), and if the developer at the time of writing their code is doing the right thing, then we should not be raising an exception simply because the state of the library may have changed outside of their control.

However, returning a `nil` listener to me is also wrong.  Subscribing to a channel that is failed is not necessarily wrong, it just means the listener will not be called until the channel is attached again.  So I don't think we should do anything if someone registers a listener for a failed channel because later, the channel could feasibly become attached if explicitly reattached.

I wonder if perhaps http://docs.ably.io/client-lib-development-guide/features/#RTL7c is simply wrong and whilst we implicitly attach, if we cannot we simply don't, or we emit an error on the channel at best.

@paddybyers @SimonWoolf WDYT?